src/Routing/RouteSubscriber.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Drupal\password_enhancements\Routing;
+
+use Drupal\config_translation\ConfigMapperManagerInterface;
+use Drupal\Core\Routing\RouteSubscriberBase;
+use Drupal\Core\Routing\RoutingEvents;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Dynamic route listener.
+ */
+class RouteSubscriber extends RouteSubscriberBase {
+
+  /**
+   * The mapper plugin discovery service.
+   *
+   * @var \Drupal\config_translation\ConfigMapperManagerInterface
+   */
+  protected $mapperManager;
+
+  /**
+   * Constructs a new RouteSubscriber.
+   *
+   * @param \Drupal\config_translation\ConfigMapperManagerInterface $mapper_manager
+   *   The mapper plugin discovery service.
+   */
+  public function __construct(ConfigMapperManagerInterface $mapper_manager) {
+    $this->mapperManager = $mapper_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    // Come after the config translation route subscriber.
+    $events[RoutingEvents::ALTER] = ['onAlterRoutes', -111];
+    return $events;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function alterRoutes(RouteCollection $collection) {
+    $mappers = $this->mapperManager->getMappers($collection);
+    $constraint = &$mappers['password_enhancements_constraint'];
+    $route = $constraint->getOverviewRoute();
+    $route->setOption('parameters', $route->getOption('parameters') + [
```

Comment:
1. `$route->getOption()` can return null, check its docblock.
2. I think https://3v4l.org/9JhKQ is not your expected result here, this is the reason why I suggested to use the `NestedArray` utility class earlier. 

```php
$params = $route->getOption('parameters') ?? [];
NestedArray::setValue($params, ['password_enhancements_policy', 'type'], 'entity:password_enhancements_policy');
```

Reply:
Ah, I see, but why not use `array_merge_recursive()` then?